src/backend/distributed/commands/alter_table.c

```diff
@@ -402,7 +415,15 @@ UndistributeTable(TableConversionParameters *params)
 	params->conversionType = UNDISTRIBUTE_TABLE;
 	params->shardCountIsNull = true;
 	TableConversionState *con = CreateTableConversion(params);
-	return ConvertTable(con);
+	TableConversionReturn *tableConversionReturn = ConvertTable(con);
```

Comment:
I thought accumulating `GenerateAlterTableAttachPartitionCommand` is the problem inside `ConvertTable`.

Is this solving that problem?

Reply:
We should accumulate only attachPartitionCommands and foreignKeyCommands now. We should refactor the alter table if we want to free those as well. 

I could not reproduce the issue linked with the PR by only leaking above mentioned commands.

Reply:
The main problem is actually where we deparse and execute commands using our utility executor method. We do not free at `ProcessUtilityParseTree` which causes huge chunk of memory to leak when we try to attach all partitions.

Reply:
Currently we only leak attachPartitionCommands and foreignKeyCommands, but if we still want to free them for each partition, then we should refactor the alter distributed table udf.

I could not reproduce the memory issue linked with the PR by only leaking above mentioned commands. It seems fine to accumulate those.